muonDIS/makeMuonDIS.py

```diff
@@ -0,0 +1,228 @@
+#!/usr/bin/env python
+"""Script to generate DIS events for muons in Pythia6, and sve to a ROOT file (along with the original muon's soft interactions)."""
+
+import argparse
+import logging
+import time
+from array import array
+
+import ROOT as r
+
+logging.basicConfig(level=logging.INFO)
+PDG = r.TDatabasePDG.Instance()
+
+
+def rotate(px, py, pz, theta, phi):
+    """Rotate the daughter particle momentum to align with respect to the muon's momentum."""
+    momentum = r.TVector3(px, py, pz)
+
+    rotation = r.TRotation()
+    rotation.RotateY(theta)  # Rotate around the Y-axis
+    rotation.RotateZ(phi)  # Rotate around the Z-axis
+
+    # Apply the rotation to the momentum vector
+    rotated_momentum = rotation * momentum
+
+    return rotated_momentum.X(), rotated_momentum.Y(), rotated_momentum.Z()
+
+
+def makeMuonDIS():
+    """Generate DIS events."""
+    parser = argparse.ArgumentParser(description=__doc__)
+    parser.add_argument("-f", "--inputFile", help="Input file to use", required=True)
+    parser.add_argument(
+        "-nJob", "--nJob", type=int, help="Process ID, gives muon index", required=True
+    )
+    parser.add_argument(
+        "-nPerJobs",
+        "--nPerJobs",
+        type=int,
+        help="The number of muons per file",
+        required=True,
+    )
```

Comment:
This maybe a bit tricky since to handle both SBT and Tr cases in the makeMuonDIS script, we adjust the generation by modifying the number of muons the script takes based on the case type. For the Tr case, where there are fewer muons, we limit the number of muons (essentially to 1 mostly), since nDISPerMuon ~1000. But for the SBT case, since we have a larger muon sample we allow more muons given that the nDISPerMuon ~100 or so. This allows for some appropriate balance in the final output as well as production time.  We  _could_ have a start event and end event instead of these, but this will require a rewrite of the existing algorithm and retesting.

Reply:
Maybe you can explain this in more detail at the CM, but this still seems like something better manages in e.g. the HTCondor submission scripts.